packages/components/src/number-control/test/index.js

```diff
@@ -91,7 +91,7 @@ describe( 'NumberControl', () => {
 			await waitFor( () => {
 				expect( spy ).toHaveBeenCalledTimes( 2 );
 				expect( spy ).toHaveBeenNthCalledWith( 1, '1' );
-				expect( spy ).toHaveBeenNthCalledWith( 2, 4 );
+				expect( spy ).toHaveBeenNthCalledWith( 2, '4' );
```

Comment:
Yep ok, I'll try to keep the changes in `TimePicker` for now.

I do think this is a mistake in `NumberControl`, though, and that `value` should always be a string. It will come up when `NumberControl` is converted to TypeScript because we'll see errors about coercing `NumberControl`'s `value: number | string` to `InputControl`'s `value: string`. `NumberControl` is supposed to be a glorified `input[type=number]` so I think it makes senes that value should work the same as it does in `HTMLInputElement`.

Reply:
I had already one attempt at typing `NumberControl`, and I have to give up mostly because of the ambiguity around which type should be used for `value` (`string`, like `HTMLInputElement`, or `number`, since it's an input field dealing with numbers?). We decided to punt that refactor forward, and instead refactor to TypeScript its dependent components: `UnitControl` (done) and `RangeControl` (WIP).

Once `RangeControl` is typed, I'll attempt the refactor once again, and I'll make sure to involve you and the rest of the folks helping this effort!